### PR TITLE
96 Login page error handling

### LINKED ISSUE #96: Error handling when submit login page with empty field

### PR DESCRIPTION
**Pull request by ttnsy**
- Closes: #96 
- Fix error when login with empty fields by adding `if (token && token !== "undefined")` in `server-session.js`
- Require email & password input 
- Process login based on API fetch status, if status is not started with `20`, it will throw message based on the API message.

### CODE REVIEW

File: src/lib/server-session.js
Comment (alfonsusac) [inline src/lib/server-session.js:1]: line 10: return is still null. return needs to be {} as specified in the task.

**Review by alfonsusac — APPROVED** [review]
Preferrably only focus one problem at a time.

You put `required` props in some of the inputs but it would be nice if we add `required` props to every form. 

It will be easier to track and i dont have to remember which <input> doesn't have a required props.

**Review by alfonsusac — CHANGES_REQUESTED** [review]
Preferrably only focus one problem at a time.

You put `required` props in some of the inputs but it would be nice if we add `required` props to every form. 

It will be easier to track and i dont have to remember which <input> doesn't have a required props.

Request Changes:
(server-session.js)
- line 10: return is still null. return needs to be {} as specified in the task.

**Comment by ttnsy** [comment]
Hi @alfonsusac,

> * line 10: return is still null. return needs to be {} as specified in the task.

See my comment regarding this [here](https://github.com/alfonsusac/devscale-eventmaker/issues/96#issuecomment-1946691600)

> You put `required` props in some of the inputs but it would be nice if we add `required` props to every form.

Can you comment on which line? There's only 2 fields in LoginForm: email & password and i set both of them to be `required`.

File: src/lib/server-session.js
Comment (ttnsy) [reply src/lib/server-session.js:1]: changing this to {} causing error in the app. can you explain why this is required?

Comment (alfonsusac) [reply src/lib/server-session.js:1]: nevermind

Comment (alfonsusac) [reply src/lib/server-session.js:1]: as long as it works, its okay i guess. I thought that it wouldnt break

**Review by alfonsusac — APPROVED** [review]

**Pull request merged**